tests/integration/mixer/policy/ratelimiting_test.go

```diff
@@ -92,30 +94,33 @@ func testRedisQuota(t *testing.T, config bookinfo.ConfigFile, destinationService
 		defer deleteConfigOrFail(t, config, ctx)
 		util.AllowRuleSync(t)
 
-		res := util.SendTraffic(ing, t, "Sending traffic...", "", "", 300)
-		_, _ = util.FetchRequestCount(t, prom, destinationService, "",
-			bookInfoNameSpaceStr, 300)
-
-		totalReqs := res.DurationHistogram.Count
-		succReqs := float64(res.RetCodes[http.StatusOK])
-		badReqs := res.RetCodes[http.StatusBadRequest]
-		actualDuration := res.ActualDuration.Seconds() // can be a bit more than requested
-
-		t.Log("Successfully sent request(s) to /productpage; checking metrics...")
-		t.Logf("Fortio Summary: %d reqs (%f rps, %f 200s (%f rps), %d 400s - %+v)",
-			totalReqs, res.ActualQPS, succReqs, succReqs/actualDuration, badReqs, res.RetCodes)
-
-		got429s, _ := util.FetchRequestCount(t, prom, destinationService, "", bookInfoNameSpaceStr,
-			300)
-		if got429s == 0 {
-			attributes := []string{fmt.Sprintf("%s=\"%s\"", util.GetDestinationLabel(),
-				util.Fqdn(destinationService, bookInfoNameSpaceStr)),
-				fmt.Sprintf("%s=\"%d\"", util.GetResponseCodeLabel(), 429),
-				fmt.Sprintf("%s=\"%s\"", util.GetReporterCodeLabel(), "destination")}
-			t.Logf("prometheus values for istio_requests_total for 429's:\n%s",
-				util.PromDumpWithAttributes(prom, "istio_requests_total", attributes))
-			t.Errorf("Could not find 429s")
-		}
+		retry.UntilSuccessOrFail(t, func() error {
+			res := util.SendTraffic(ing, t, "Sending traffic...", "", "", 300)
+			totalReqs := res.DurationHistogram.Count
+			succReqs := float64(res.RetCodes[http.StatusOK])
+			badReqs := res.RetCodes[http.StatusBadRequest]
+			actualDuration := res.ActualDuration.Seconds() // can be a bit more than requested
+
+			t.Log("Successfully sent request(s) to /productpage; checking metrics...")
+			t.Logf("Fortio Summary: %d reqs (%f rps, %f 200s (%f rps), %d 400s - %+v)",
+				totalReqs, res.ActualQPS, succReqs, succReqs/actualDuration, badReqs, res.RetCodes)
+
+			// We expect to receive 50 429's as the rate limit is set to allow 50 requests in 30s.
```

Comment:
Do we expect to receive 250 or 50 429?

Reply:
50 429's... and 250 200's

Reply:
I might miss something here.. By 50 requests is allowed in 30s, doesn't it mean that 50 requests will get 200, and rest will just be denied with 429?

Reply:
Thanks for catching!! Fixed